src/main/scala/metadata.scala

```diff
@@ -0,0 +1,83 @@
+package suffuse
+
+/** AKey is an attribute key: a Metadata map key which carries
+ *  both the type of the corresponding value, and a value to
+ *  provide when there is no value in the map. This as opposed
+ *  to wrapping everything in sight in Option.
+ */
+final class AKey[A](description: String, empty: => A) {
+  def emptyValue: A = empty
+  override def toString = description
+}
+
+/** Attr is a dependently typed key/value pair.
+ */
+trait Attr extends Any {
+  type Type
+  def key: AKey[Type]
+  def value: Type
+
+  def pair                               = key -> value
+  def matches[A](implicit that: AKey[A]) = key == that
+  def to_s                               = s"$key: $value"
```

Comment:
In psp-std `ShowDirect` is a way of obtaining a `Show[A]` typeclass without having to also define an implicit. Object's `toString` is supposed to be used only for debugging and `to_s` is supposed to be principled, so there's no reason to conflate these up front. In addition, overriding toString subverts any synthetic toString, such as the one produced by case classes. There's no way to recover the synthetic implementation once a toString override appears in the inheritance chain.